### PR TITLE
Snapshot temporary directory support configurable

### PR DESCRIPTION
**Pull request by qiujiayu**
### Motivation:

#932 

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java [hunk 1]
```diff
@@ -78,14 +81,33 @@ public void setServerAddr(Endpoint addr) {
...
+            this.tempPath = buildTempPath(this.path);
+        } else {
+            File pathFile = new File(path);
+            File tempPathFile = new File(tempPath);
+
+            String pathAbsolutePath = pathFile.getAbsolutePath();
+            String tempPathAbsolutePath = tempPathFile.getAbsolutePath();
+            if (pathAbsolutePath.equals(tempPathAbsolutePath) || pathAbsolutePath.startsWith(tempPathAbsolutePath)) {
+                this.tempPath = buildTempPath(this.path);
+                LOG.info("The snapshot temp path is {}", tempPath);
```
Comment (killme2008) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java:97]: The log can be moved outside. This info may be useful for the users.

File: jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java [hunk 2]
```diff
@@ -108,13 +130,12 @@ public boolean init(final Void v) {
 
         // delete temp snapshot
         if (!this.filterBeforeCopyRemote) {
-            final String tempSnapshotPath = this.path + File.separator + TEMP_PATH;
-            final File tempFile = new File(tempSnapshotPath);
+            final File tempFile = new File(tempPath);
```
Comment (killme2008) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java:133]: `this.tempPath`, always prefix `this.` for instance variables.

File: jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java [hunk 3]
```diff
@@ -108,13 +130,12 @@ public boolean init(final Void v) {
...
             if (tempFile.exists()) {
                 try {
                     FileUtils.forceDelete(tempFile);
                 } catch (final IOException e) {
-                    LOG.error("Fail to delete temp snapshot path {}.", tempSnapshotPath, e);
+                    LOG.error("Fail to delete temp snapshot path {}.", tempPath, e);
```
Comment (killme2008) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotStorage.java:138]: The same as above.

**Review by killme2008 — COMMENTED** [review]
Good job, thanks. There are some minor comments.

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by fengjiachun** [comment]
Thanks @qiujiayu

**Pull request merged**